ToolBox/SECDataDownloader/SECDataConverter.cs

```diff
@@ -65,247 +144,283 @@ public class SECDataConverter
+                        if (currentTagName == "PAPER")
                         {
-                            symbol = new List<string>();
-                            CikTicker[cikFormatted] = symbol;
+                            continue;
                         }
 
-                        symbol.Add(tickerCik[0]);
-                    }
-                );
-            
-            // Merge both data sources to a single CIK -> List{T} of tickers
-            rankAndFileData.Split('\n')
-                .Select(x => x.Split('|'))
-                .ToList()
-                .ForEach(
-                    tickerInfo =>
-                    {
-                        var companyCik = tickerInfo[0].PadLeft(10, '0');
-                        var companyTicker = tickerInfo[1].ToLower();
+                        // Don't encode the closing tag
+                        if (currentTagName == "/TEXT")
+                        {
+                            parsingText = false;
+                        }
 
-                        List<string> symbol;
-                        if (!CikTicker.TryGetValue(companyCik, out symbol))
+                        // To ensure that we can serialize/deserialize data with hours, minutes, seconds
+                        if (currentTagName == "FILING-DATE" || currentTagName == "PERIOD" ||
+                            currentTagName == "DATE-OF-FILING-CHANGE" || currentTagName == "DATE-CHANGED")
                         {
-                            symbol = new List<string>() {companyTicker};
-                            CikTicker[companyCik] = symbol;
+                            newTextLine = $"{newTextLine.TrimEnd()} 00:00:00";
                         }
-                        else if (!symbol.Contains(companyTicker))
+
+                        // Encode all contents inside tags to prevent errors in XML parsing.
+                        // The json deserializer will convert these values back to their original form
+                        if (!parsingText && HasValue(newTextLine))
                         {
-                            symbol.Add(companyTicker);
+                            newTextLine =
+                                $"<{currentTagName}>{SecurityElement.Escape(GetTagValueFromLine(newTextLine))}</{currentTagName}>";
+                        }
+                        // Escape all contents inside TEXT tags
+                        else if (parsingText)
+                        {
+                            newTextLine = SecurityElement.Escape(newTextLine);
                         }
-                    }
-                );
 
-            KnownEquities = Directory.GetFiles(knownEquityFolder)
-                .Select(x => Path.GetFileNameWithoutExtension(x).ToLower())
-                .ToList();
-        }
+                        // Don't encode the opening tag
+                        if (currentTagName == "TEXT")
+                        {
+                            parsingText = true;
```

Comment:
nit - more succinctly as `parsingText = currentTagName == "TEXT"`

Reply:
I wish I could do so, but we need to ensure that when this advances to the next line, that the boolean value remains unchanged unless the tag encountered == `</TEXT>`. 

By doing as you've said, on the next line following a `<TEXT>` tag, the condition `parsingText` would be set to false, and we'd stop escaping the inputs.